src/main/groovy/com/vherasymenko/avro/decoder/core/binary/AvroBinaryDecoderService.groovy

```diff
@@ -0,0 +1,22 @@
+package com.vherasymenko.avro.decoder.core.binary
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.io.DatumReader
+import org.apache.avro.io.DecoderFactory
+
+/**
+ * The avro binary decoder.
+ */
+class AvroBinaryDecoderService implements AvroBinaryDecoderPort {
+
+    @Override
+    void decodeEvent( Schema schema, byte[] event ) {
+        def decoder = new DecoderFactory().binaryDecoder( event, null )
```

Comment:
When you do your research, don't forget to find out if the codecs are thread-safe or not.  If they are, we can inject them as beans, otherwise we'll have to continue to create them as needed.